nxdlTypes.xsd

```diff
@@ -519,6 +523,48 @@
 			" />
 	</xs:simpleType>
 	
+	<xs:complexType name="NX_CCOMPLEX">
+		<xs:annotation>
+			<xs:documentation>Compound type cartesian representaion of complex numbers (real and imaginary parts) in NeXus</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="real" type="xs:double"/>
+			<xs:element name="imaginary" type="xs:double"/>
+		</xs:sequence>
+	</xs:complexType>
+	
+	<xs:complexType name="NX_PCOMPLEX">
+		<xs:annotation>
+			<xs:documentation>Compound type polar representaion of complex numbers (amplitude and phase) in NeXus</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="amplitude" type="xs:double"/>
+			<xs:element name="phase" type="xs:double"/>
+		</xs:sequence>
+	</xs:complexType>
+	
+	<xs:simpleType name="NX_COMPLEX">
+		<xs:annotation>
+			<xs:documentation>Compound type representaion of complex numbers (real and imaginary parts) in NeXus</xs:documentation>
+		</xs:annotation>
+		<xs:union memberTypes="
+			nxdl:NX_CCOMPLEX  
+			nxdl:NX_PCOMPLEX
+			" />
+	</xs:simpleType>
+	
+	<xs:complexType name="NX_QUATERNION">
+		<xs:annotation>
+			<xs:documentation>Compound type representaion of quaternion numbers (scalar,i,j,k) in NeXus</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="scalar" type="xs:double"/>
```

Comment:
I prefer `real` here as it mirrors the complex number definition and a scalar/vector interpretation of a quarternoin may be confusing things by introducing a geometric meaning.

Reply:
done